src/components/StyledInput.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { FC, HTMLAttributes, useState } from "react";
+
+import { StyleSheet, css } from "aphrodite";
+import { colors } from "styles/palette";
+
+interface Props {
+  value: string;
+  error?: string;
+  focused?: boolean;
+  onBlur?: () => void;
+  onFocus?: () => void;
+  label?: string;
+  placeholder?: string;
+  style?: HTMLAttributes<HTMLInputElement>;
+  type?: "submit" | "text" | "number";
+}
+
+const StyledInput: FC<Props> = ({
+  error,
+  focused = false,
+  label,
+  onBlur,
+  onFocus,
+  placeholder,
+  style,
+  type = "text",
+  value = "",
+}) => {
+  const [inputValue, setInputValue] = useState(value);
+  const [isFocused, setIsFocused] = useState(focused);
+
+  const handleFocus = () => {
+    setIsFocused(true);
+    if (onFocus) {
+      onFocus();
+    }
+  };
+  const handleBlur = () => {
+    setIsFocused(false);
+    if (onBlur) {
+      onBlur();
+    }
+  };
+
+  return (
+    <div>
+      <label htmlFor={label}>
+        {label}
+        <input
+          id={label}
+          onFocus={handleFocus}
+          onChange={(event) => setInputValue(event.target.value)}
```

Comment:
onChange={onChange}
 
 and this handler should be passed from the parent component. Also move `const [inputValue, setInputValue] = useState(value);` into parent